design/stackhub.md

```diff
@@ -0,0 +1,207 @@
+
+You can either add the required Tekton Trigger and Bindings configuration to your existing CollectionHub before you upgrade, or one can move directly to a StackHub with separate pipeline release.
+
+Please note that this transition will also require that any  Github webhooks that were previously established with the Tekton dashboard will need to be reestablished.
+
+
+#### (Kabanero updates from 0.1.n-0.2.n)
+
+No support for directly upgrading from these releases.
+
+
+##  Dependent Projects
+
+### Appsody
+
+This design will ask the Appsody open source community to extract some additional metadata during StackHub build (that was previously captured in the Kabanero Collection Hub build), which exposes the configured registry and org names in the `index.yaml.`
+
+Although not required for this design change in Kabanero, the enterprise governance lifecycle would improve greatly for both Appsody and Kabanero from an improvement in StackHub defintion allowing by-reference stack specification when constructing builds. 
+
+## Kabanero Collections
+
+This repository will live on to provide service for previous levels of Kabanero for a period of time.
+
+A new `0.5.n` release of the Collection Hub will be produced to preserve the UBI stack variants until they can find a home elsewhere, and to update them with current maintainence.  
+
+## Kabanero Operator
+
+Most of this specification applies to the Kabanero operator, but specifically:
+
+- **kabanero.yaml**  
+  - The default Kabanero CR will point at the Appsody Stackhub by default, configuring the default pipelines `0.6.0` release.
+  Changing the default Kabanero CR to point at an Appsody stack-hub release, the containers that are installed by Kabanero by example will be the Ubuntu variants.  
+  - A new example of the Kabanero CR will be added that will point at the new `0.5.n` collection hub for those consumers that wish to continue to use the UBI variants of the stacks.
+
+
+## Kabanero Pipelines
+
+Today the pipelines that are included in the Collection Hub are cut-and-pasted into the kabanero/collections repository.  With the elimination of the Collection Hub and Collections, the Kabanero CR will be configured with pipelines, and we need to release them independently.
+
+Since the pipelines can contain samples, we should isolate them from the pipelines we want to have as best practices.
+
+Until Kabanero eventing is rationalized and generally available, Kabanero pipeline step ensures that relevant pipelines (e.g. build) are operating against Collections are active.   This logic interacts with the Collections CRs and will need to change to interact with the Stack CRs.
+
+## Kabanero CLI and REST APIs
+
+The Kabanero CLI has methods for interacting with collections and collection hub.  
+
+- **kabanero deactivate** - Remove the specified collection from the list of available application types, without deleting it from the Kabanero instance.
```

Comment:
I think this should be deprecated and all changes to the active state of the instance be made through the Kabanero CRD.

Reply:
Again, these CLIs are generally used by Champ during the development of enterprise, curated stacks, not in production clusters.  I think we could look at doing some additional clarification as part of follow-on work.